test/object-store/sync/client_reset.cpp

```diff
@@ -237,8 +237,12 @@ TEST_CASE("sync: client reset", "[client reset]") {
         ++after_callback_invocations;
         REQUIRE(before);
         REQUIRE(before->is_frozen());
-        REQUIRE(before->read_group().get_table("class_object"));
-        REQUIRE(before->config().path == local_config.path);
+        auto before_schema_mode = before->config().schema_mode;
+        if (before_schema_mode != SchemaMode::AdditiveExplicit &&
+            before_schema_mode != SchemaMode::AdditiveDiscovered) {
+            REQUIRE(before->read_group().get_table("class_object"));
+            REQUIRE(before->config().path == local_config.path);
+        }
```

Comment:
This is the reason: 
```
  REQUIRE( before->read_group().get_table("class_object") )
  due to unexpected exception with message:
   Unsupported schema changes were made by another client or process. For a
   synchronized Realm, this may be due to the server reverting schema changes
   which the local user did not have permission to make.
  - Property 'preexisting table name.remotely added property' has been removed.
  - Property 'same new table name.remotely added property' has been removed.
 ```
Essentially for additive schemas I am now appending the mismatches, which will trigger an exception to be thrown if the table will access.

Reply:
@ironage scratch this, it was my wrong way of populating the schema that was actually breaking the test. So it was my mistake.